### PR TITLE
using qcylindermesh instead of OFFmesh for cylinders

### LINKED ISSUE #397: Render cylinders as cylinders in the 3D view
Instead of generating a mesh for cylinders, use the higher level interface of qt3d and simply tell it to render a cylinder.

The `off_geometry` method can be removed from `OFFGeometry` and `CylindricalGeometry`.

### PR DESCRIPTION
**Pull request by rerpha**
### Issue

Closes #397 
### Description of work

Only spent a few mins on this so if it causes issues i'll come back to it at a later date.
This uses `QCylinderMesh`es instead of drawing cylinders as OFF meshes in the qt3d view. This may give some performance benefits, but mainly it reduces a lot of code that we have to look after (converting cylinders to off meshes) 

### Acceptance Criteria 

Cylinders should be added as usual in the 3d view and the same output should be shown in the nexus file and tree view. 

### UI tests

*How you have updated the UI tests document to fit your changes*

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

File: nexus_constructor/geometry/off_geometry.py [hunk 1]
```diff
@@ -38,7 +39,7 @@ def winding_order_indices(self) -> List[int]:
 
     @property
     @abstractmethod
-    def off_geometry(self) -> "OFFGeometry":
+    def mesh(self) -> OffMesh:
```
Comment (matthew-d-jones) [inline nexus_constructor/geometry/off_geometry.py]: I think the return type should be `Union(Qt3DExtras.QCylinderMesh, OFFMesh)`

Comment (matthew-d-jones) [reply nexus_constructor/geometry/off_geometry.py]: Or if `OFFMesh` and `QCylinderMesh` inherit from a base mesh class then that would be appropriate

File: nexus_constructor/off_renderer.py [hunk 2]
```diff
@@ -7,11 +7,14 @@
...
 from nexus_constructor.pixel_data import PixelData, PixelGrid
-from nexus_constructor.geometry import OFFGeometry
 from PySide2.Qt3DRender import Qt3DRender
 from PySide2.QtGui import QVector3D
 import struct
 import itertools
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:  # pragma: no cover
+    from nexus_constructor.geometry.off_geometry import OFFGeometry
```
Comment (matthew-d-jones) [inline nexus_constructor/off_renderer.py]: As you've changed to strings for the OFFGeometry type hint, surely this import doesn't do anything?

Comment (rerpha) [reply nexus_constructor/off_renderer.py]: this is to suppress flake8 warnings, but shouldn't need to exist if i make them inherit from a base class

**Comment by rerpha** [comment]
I am a bit stuck with this, have tried a few different methods such as bundling in a transform with the cylinder, but i'm not sure how to get it finished. I am going to shelve for now.

**Comment by rerpha** [comment]
Closing for now as the branch will diverge with upcoming changes and I can't find a straightforward way of fixing this.

**Pull request closed without merging**